lib/cuke_linter/linters/unique_scenario_names_linter.rb

```diff
@@ -0,0 +1,108 @@
+module CukeLinter
+  # A linter that detects non-unique scenario names
+  class UniqueScenarioNamesLinter < Linter
+
+    def initialize
+      super
+      @scenario_names = {}
+    end
+
+    def rule(model)
+      return nil unless valid_model?(model)
+
+      feature_file = model.get_ancestor(:feature_file)
+      return nil if feature_file.nil?
+      
+      file_path = feature_file.path
+
+      case model
+      when CukeModeler::Rule
+        check_rule(model, file_path)
+      when CukeModeler::Outline
+        check_scenario_outline(model, file_path)
+      when CukeModeler::Scenario
+        check_scenario(model, file_path)
+      else
+        nil
+      end
+    end
+
+    private
+
+    def check_rule(model, file_path)
+      problems = []
+      rule_key = "#{file_path}:#{model.name}"
+
+      model.scenarios.each do |scenario|
+        problem = check_scenario(scenario, rule_key)
+        problems << problem if problem
+      end
+
+      model.outlines.each do |outline|
+        problem = check_scenario_outline(outline, rule_key)
+        problems << problem if problem
+      end
+
+      problems.first
+    end
+
+    def check_scenario(model, scope_key)
+      scenario_name = model.name
+      check_duplicate(scenario_name, scope_key)
+      return nil unless duplicate_name?(scenario_name, scope_key)
+      
+      @message = 'Scenario names are not unique'
+      build_problem(model)
```

Comment:
build_problem is automatically calling in the `lint` method of Linter (see `lib/cuke_linter/linters/linter.rb`). So you don't need to call it yourself

Reply:
Right! Fixed